jwst/outlier_detection/outlier_detection.py

```diff
@@ -194,11 +194,16 @@ def do_detection(self):
             for model in drizzled_models:
                 if save_intermediate_results:
                     log.info("Writing out resampled exposures...")
-                    self.save_model(
-                        model,
-                        output_file=model.meta.filename,
-                        suffix=self.resample_suffix
+                    self.save_results = True
+                    self.output_use_model = model.meta.filename
+                    self.log = log
+                    Step.save_model(
+                        self,
+                        model=model,
+                        output_file=model.meta.filename.lstrip("_"),
+                        suffix=self.resample_suffix.lstrip("_").replace('.fits','')
```

Comment:
Typically the inputs to `outlier_detection` are individual cal products with file names like "jw00632011001_04102_00001_nrca3_cal.fits". If `lstrip("_")` removes everything to the left of the first occurrence of "_" in that name, won't it be left with just "jw00632011001"? Or does it work from the last occurrence of "_"? I guess ultimately the desired file name is "jw00632011001_04102_00001_nrca3_i2d.fits", right? Or something else, cuz I guess that could conflict with existing files of that same name that were created by the `calwebb_image2` pipeline.

Reply:
Ahhh yes. Trying to do saving while not in a Step object. The solution should be just below this code section. In the master branch, look to line 219.